lib/film.rb

```diff
@@ -6,10 +6,10 @@ def self.add(userId, filmId)
     rs = DatabaseConnection.query("INSERT INTO userFilms ('userId', 'filmId') VALUES (#{userId}, #{filmId}) RETURNING *;")
   end
 
-  def self.all(user_id)
-    rs = DatabaseConnection.query("SELECT * FROM usersFilms WHERE userId=#{user_id}")
+  def self.findbyuserid(userId)
+    rs = DatabaseConnection.query("SELECT * FROM films FULL OUTER JOIN usersFilms ON films.filmid = usersFilms.filmid WHERE userid = #{userId} ")
     rs.map do |row|
-      row[:filmId]
+      {filmId: row[:filmId], userId: row[:userId], year: row[:year], title: row[:title], posterPath: row[:posterPath]}
```

Comment:
Can we change this line to:

Film.new(rs[0]["id"].to_i, rs[0]["filmid"].to_i, rs[0]["title"], rs[0]["posterpath"], rs[0]["year"].to_i)